lookout/style/typos/model.py

```diff
@@ -0,0 +1,112 @@
+from typing import Union, List
+from collections import defaultdict
+
+import pandas
+import modelforge
+
+from lookout.core.analyzer import AnalyzerModel
+from lookout.style.typos.token_parser import TokenParser
+from lookout.style.typos.correction_model import ContextXgbModel
+from lookout.style.typos.typos_functions import flattify_data
+
+
+class TyposModel(AnalyzerModel):
+    NAME = "typos"
+    VENDOR = "source{d}"
+
+    N_CANDIDATES = 3
+    CONFIDENCE_THRESHOLD = 0
+
+    def __init__(self, confidence_threshold: float=CONFIDENCE_THRESHOLD,
+                 n_candidates: int=N_CANDIDATES):
+        super().__init__()
+        print(self.meta)
+        self.confidence_threshold = confidence_threshold
+        self.n_candidates = n_candidates
+        self.parser = TokenParser()
+
+    def check_identifiers(self, identifiers: List[str],
+                          checker: Union[ContextXgbModel, None]) -> dict:
+        splits = []
+        for identifier in identifiers:
+            split = []
+            for token in self.parser.split(identifier):
+                split.append(token)
+            splits.append(' '.join(split))
+
+        test_df = pandas.DataFrame(columns=["node_id", "token_split"])
+        test_df["node_id"] = range(len(identifiers))
+        test_df["token_split"] = splits
+        test_df = flattify_data(test_df, new_column_name="typo")
+
+        test_df.to_csv("test_df.csv")
+
+        suggestions = {}
+        if checker is not None:
+            suggestions = checker.suggest_df(test_df,
+                                             n_candidates=3,
+                                             return_all=False)
+
+        suggestions = self.filter_suggestions(test_df, suggestions)
+        return self.group_by_id(test_df, suggestions)
+
+    def filter_suggestions(self, test_df: pandas.DataFrame,
+                           suggestions: pandas.DataFrame) -> dict:
+        """
+        Filter suggestions based on the repo specifics and
+        confidence threshold
+        :param suggestions: dict of suggestions for tokens
+               inside identifier
+        :param test_df: pandas.DataFrame with info about
+               tested identifiers
+        :return: dict of filtered suggestions
+        """
+        filtered_suggestions = {}
+        tokens = test_df.typo
+        for id, suggestion in suggestions.items():
```

Comment:
id is a bad name because it hides python `id` build-in. use `i` or `index`, for example.